jwst/tweakreg/wcsimage.py

```diff
@@ -655,43 +658,19 @@ def _calc_cat_convex_hull(self):
             # no points
             raise RuntimeError("Unexpected error: Contact sofware developer")
 
-        elif len(x) == 1:
-            # one point. build a small box around it:
-            x, y = convex_hull(x, y, wcs=self.det_to_tanp)
-
-            xv = [x[0] - 0.5, x[0] - 0.5, x[0] + 0.5, x[0] + 0.5, x[0] - 0.5]
-            yv = [y[0] - 0.5, y[0] + 0.5, y[0] + 0.5, y[0] - 0.5, y[0] - 0.5]
-
-            ra, dec = self.tanp_to_world(xv, yv)
-
-        elif len(x) == 3:
-            # two points. build a small box around them:
-            x, y = convex_hull(x, y, wcs=self.det_to_tanp)
-
-            vx = -(y[1] - y[0])
-            vy = x[1] - x[0]
-            norm = 2.0 * np.sqrt(vx * vx + vy * vy)
-            vx /= norm
-            vy /= norm
-
-            xv = [x[0] + vx, x[1] + vx, x[1] + vx, x[0] - vx, x[0] + vx]
-            yv = [y[0] + vy, y[1] + vy, y[1] + vy, x[0] - vx, x[0] + vx]
-
-            ra, dec = self.tanp_to_world(xv, yv)
-
-        else:
+        if len(x) > 2:
```

Comment:
So what happens when len(x) <= 2 and this code block doesn't get executed?

Reply:
When `len(x) <= 2` footprint (already) calculated based on image size/shape is used. This will lead to poor estimates of "overlapping" but it should work.

Reply:
👍